ksmt-core/src/main/kotlin/org/ksmt/solver/util/KExprIntInternalizerBase.kt

```diff
@@ -0,0 +1,193 @@
+package org.ksmt.solver.util
+
+import org.ksmt.expr.KExpr
+import org.ksmt.expr.transformer.KTransformerBase
+import org.ksmt.solver.util.KExprLongInternalizerBase.Companion.NOT_INTERNALIZED
+import org.ksmt.sort.KSort
+
+/**
+ * Specialized version of [KExprConverterBase] for Int native expressions.
+ * */
+abstract class KExprIntInternalizerBase : KTransformerBase {
+    @JvmField
+    val exprStack = arrayListOf<KExpr<*>>()
+
+    /**
+     * Return internalized expression or
+     * [NOT_INTERNALIZED] if expression was not internalized yet.
+     * */
+    abstract fun findInternalizedExpr(expr: KExpr<*>): Int
+
+    abstract fun saveInternalizedExpr(expr: KExpr<*>, internalized: Int)
+
+    fun <S : KSort> KExpr<S>.internalizeExpr(): Int {
+        exprStack.add(this)
+
+        while (exprStack.isNotEmpty()) {
+            val expr = exprStack.removeLast()
+
+            val internalized = findInternalizedExpr(expr)
+            if (internalized != NOT_INTERNALIZED) continue
+
+            expr.accept(this@KExprIntInternalizerBase)
+        }
+
+        return findInternalizedExpr(this).also {
+            check(it != NOT_INTERNALIZED) { "expr is not properly internalized: $this" }
+        }
+    }
+
+    inline fun <S : KExpr<*>> S.transform(operation: () -> Int): S = also {
+        saveInternalizedExpr(this, operation())
+    }
+
+    inline fun <S : KExpr<*>> S.transform(
+        arg: KExpr<*>,
+        operation: (Int) -> Int
+    ): S = also {
+        val internalizedArg = findInternalizedExpr(arg)
+
+        if (internalizedArg == NOT_INTERNALIZED) {
+            exprStack.add(this)
+            exprStack.add(arg)
+        } else {
+            val internalized = operation(internalizedArg)
+            saveInternalizedExpr(this, internalized)
+        }
+    }
+
+    inline fun <S : KExpr<*>> S.transform(
+        arg0: KExpr<*>,
+        arg1: KExpr<*>,
+        operation: (Int, Int) -> Int
+    ): S = also {
```

Comment:
Duplicated code fragment (18 lines long) here and in KExprLongInternalizerBase and in functions below. Seems like it could be extracted if `Int` and `Long` would be a generic parameter